### PR TITLE
add States component

### PR DESCRIPTION
**Pull request by gmzjuliana**
## Description
Adds States component as described in Notion: https://www.notion.so/buffer/WIP-States-7508ca11ed5946f8b8c67f2947ade44a

## Screenshots (if appropriate):
image
image
image
image
image
image
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style and guidelines of this project. 
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the **CHANGELOG** document.
- [x] I have added tests to cover my changes.
- [x] I have performed a self-review of my own code
- [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] All new and existing tests passed.
- [x] I have performed the accessibility audit of my UI changes according to the accessibility doc. 
- [ ] [Buffer Engineers] Someone from the Design team reviewed and approved my changes
- [ ] [Buffer Engineers] I have notified the BDS team of my changes in the #proj-design-system Slack channel

### CODE REVIEW

File: src/components/States/States.jsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+};
+
+Media.defaultProps = {};
+
+const Header = ({ children }) => {
+  const size = useStatesContext();
+  return (
+    <HeaderWrapper>
+      {size === 'large' && <Text type="h2">{children}</Text>}
+      {size === 'small' && <Text type="h3">{children}</Text>}
```
Comment (gmzjuliana) [inline src/components/States/States.jsx:120]: this is a bit of intended rigidity since the designs specify that these will always be either h2 or h3. Keen how this feels to y'all or if we should move to make it more flexible, rendering the children that get passed (and could be anything then)

File: src/components/States/States.jsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  children: PropTypes.node.isRequired, // fix this prop type later
+};
+
+Header.defaultProps = {};
+
+const Description = ({ children }) => {
+  const size = useStatesContext();
+  return (
+    <DescriptionWrapper size={size}>
+      <Text type="p">{children}</Text>
```
Comment (gmzjuliana) [inline src/components/States/States.jsx:135]: same rigidity as mentioned above!

**Review by moreofmorris — APPROVED** [review]
Hey there @gmzjuliana! From a design implementation point of view these look great. The way you've implemented this looks to be quite flexible for change too which I think is good. These states were designed to (hopefully) cover a lot of places we'll need and bring some consistency and simplicity (that's absolutely what these were optimised for). As we start to work on the Onboarding project more needs may arise but as far as what was designed for the moment and what you've implemented it looks like you've done it all great. Thank you! 😊 

To recap here, to implement these on the code side to be flexible for change feels great:
- Illustrations and the animations may change in the new year as Julia wants to ensure all illustrations are modern and consistent, on going chats there!
- As more use cases arise we may need to introduce other variants but I feel confident that they'll mostly be an image/media, header, text and buttons or some combination of those so flexibility there feels key

With that in mind, it looks like you've achieved this 😊 Thanks again 👍

File: src/components/States/States.jsx (hunk 1 above)
Comment (federicoweber) [reply src/components/States/States.jsx:120]: I love this @gmzjuliana, it's the perfect compromise, it allows us to pass down custom text styling using `span` but will still enforce the semantic structure.
🌮

**Review by dace — APPROVED** [review]
Thanks @gmzjuliana ! LGTM 🙌🏽 🙏🏽
